apps/root/src/services/aggregatorService.ts

```diff
@@ -177,7 +177,7 @@ export default class AggregatorService {
       hasEnoughForSwap = sortedOptions.some((option) => balance >= option.sellAmount.amount);
     }
 
-    if (usePermit2 && from.address === protocolToken.address && takerAddress && hasEnoughForSwap) {
+    if (usePermit2 && from.address === protocolToken.address && takerAddress && hasEnoughForSwap && isOnNetwork) {
```

Comment:
We agreed to implement validations no matter isOnNetwork value. Maybe we can merge this (as it's stable), and create a new task for it? @FiboApe